lib/Common/Memory/MarkContext.h

```diff
@@ -102,6 +106,7 @@ class MarkContext
     Recycler * recycler;
     PagePool * pagePool;
     PageStack<MarkCandidate> markStack;
+    PageStack<IRecyclerVisitedObject*> preciseStack;
```

Comment:
Consider wrapping this in #ifdef RECYCLER_VISITED_HOST?

Reply:
Yes, OK.